source/uri-options/tests/tls-options.json

```diff
@@ -0,0 +1,82 @@
+{
+  "tests": [
+    {
+      "description": "Valid required tls options are parsed correctly",
+      "uri": "mongodb://example.com/?tls=true&tlsCAFilePath=ca.pem&tlsClientCertFilePath=cert.pem&tlsClientKeyPassword=hunter2",
+      "valid": true,
+      "warning": false,
```

Comment:
Should this raise a warning as well since these files don't exist/aren't readable? Or maybe we can add dummy files to the test artifacts?

Reply:
Fair enough. Looks like there needs to be better separation of URI parsing and URI validation in the Python driver.

Reply:
I think don't think it's necessarily an issue for the Python driver to eagerly check the files (although it will make the testing a bit more difficult); I mostly just don't want to mandate that all drivers have to do it eagerly

Reply:
I agree that parsing a URI should be separate from verifying whether a file that is referenced exists (though as a user I would certainly like the driver to inform me with a warning or an error when a file I specified does not exist).

Reply:
Yep! The test isn't saying that the driver can't warn at all, just that it doesn't need to be done as part of URI parsing. I'm guessing that most drivers already raise an error in the code path where the file is actually read (i.e. when the TLS connection is initiated), so forcing them to add code to check for the file and raise a warning before this seems redundant at best.

Reply:
I think the problem @prashantmital might be hitting is the Python driver validates that the files exist and are readable. Another tricky part of this is the file paths are obviously different on Unix and Windows.

Adding empty files to the test suite seems reasonable, since we won't need an absolute file path (avoids the Windows problem) and the contents don't matter (we're testing parsing the options, not validating their contents). That would solve the Python problem and any other drivers that do the same sort of validation. There doesn't need to be a mandate that all drivers do that extra validation PyMongo does.

Reply:
I would be in favor of adding in the empty files like @behackett suggested.

Reply:
I'll add them